### PR TITLE
master: fix unit test for OfflineMember

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 

~~when master receives GRPC request, first will execute `isLeaderAndNeedForward` to judge whether this master is the leader, and will use the `leaderClient` to forward request to the leader. But the `leaderClient` is not protected by lock, it may be close when sending requests.~~

the unit test for `OfflineMember` may failed, refine it

### What is changed and how it works?

~~add lock~~

make sure `s3` is not leader, and then offline it.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by lance6716** [comment]
LGTM

**Comment by lance6716** [comment]
suspect these's bug in test `ha_master` 😢 timeout for 2 times

**Comment by WangXiangUSTC** [comment]
> suspect these's bug in test `ha_master` 😢 timeout for 2 times

I will check it

File: dm/master/server_test.go [hunk 1]
```diff
@@ -1287,7 +1287,14 @@ func (t *testMaster) TestOfflineMember(c *check.C) {
 
 	cancel3()
 	s3.Close()
-	time.Sleep(5 * time.Second)
+	// make sure s3 is not the leader, otherwise OfflineMember request will send to s3
```
Comment (WangXiangUSTC) [inline dm/master/server_test.go:1290]: @lance6716 the unit test may run failed because s3 is the leader

**Comment by WangXiangUSTC** [comment]
It may cause deadlock if add lock for request. For example, A receives request and add lock, and B is leader, so A  forward request to B. When B receives request, the A becomes leader, so B forward request to A, but A is locked. So I revert the change in 5b32952. It doesn't matter without lock, just return an error if the leader connection changed, and the user can retry. @GMHDBJD @lance6716

**Pull request merged**